### PR TITLE
chore(screenshotter): support Browserstack and test on Safari 13.1

### LINKED ISSUE #1859: Test on latest browsers and Safari
#1842 has shown that KaTeX may break on new browsers. We should consider testing on latest browsers.

### PR DESCRIPTION
**Pull request by ylemkimon**
Closes #1859.

### CODE REVIEW

File: dockers/screenshotter/screenshotter.sh [hunk 1]
```diff
@@ -17,7 +17,7 @@ cleanup() {
 container=
 trap cleanup EXIT
 status=0
-for browserTag in firefox:2.48.2 chrome:2.48.2; do
+for browserTag in firefox:3.141.59-20200525 chrome:3.141.59-20200525; do
```
Comment (ylemkimon) [inline dockers/screenshotter/screenshotter.sh:20]: I forgot to change these in #2304.

**Comment by ylemkimon** [comment]
@kevinbarabash 
> Do we only upload artifacts when a screenshot test fails?

Yes, it does.

File: .circleci/config.yml [hunk 2]
```diff
@@ -107,3 +130,17 @@ workflows:
       - screenshotter:
           name: chrome
           browser: chrome
+      - screenshotter:
+          name: safari
+          browser: node
+          browserstack: |
+              {
+                "browserName": "Safari",
+                "browser_version": "13.1",
```
Comment (kevinbarabash) [inline .circleci/config.yml:139]: Is the mix of snake case and camel case intentional here?

File: .circleci/config.yml [hunk 3]
```diff
@@ -103,16 +115,27 @@ jobs:
...
+      - when:
+          condition: << parameters.browserstack >>
+          steps:
+            - screenshot:
+                flags: --browserstack --selenium-capabilities '<< parameters.browserstack >>'
+      - unless:
+          condition: << parameters.browserstack >>
+          steps:
+            - screenshot:
+                flags: --selenium-ip localhost
```
[the remark this answers is not in the document]
Comment (ylemkimon) [reply .circleci/config.yml:114]: @kevinbarabash One is `when` (equivalent to `if`), and the other is `unless` (equivalent to `if not`).

File: dockers/screenshotter/screenshotter.js [hunk 4]
```diff
@@ -81,6 +86,24 @@ let katexURL = opts.katexUrl;
...
+let bsLocal;
+if (opts.browserstack) {
+    // https://www.browserstack.com/automate/node
+    if (!seleniumURL) {
+        seleniumURL = "http://hub-cloud.browserstack.com/wd/hub";
+    }
+    // https://www.browserstack.com/local-testing/automate#test-localhost-websites
+    if (!katexIP && opts.browser === "safari") {
+        katexIP = "bs-local.com";
+    }
```
[the remark this answers is not in the document]
Comment (ylemkimon) [reply dockers/screenshotter/screenshotter.js:98]: @kevinbarabash One may use proxy or Browserstack-compatible service (I don't know if there is any).

File: .circleci/config.yml (hunk 2 above)
Comment (ylemkimon) [reply .circleci/config.yml:139]: @kevinbarabash `browserName` is [an official selenium capability](https://github.com/SeleniumHQ/selenium/wiki/DesiredCapabilities#used-by-the-selenium-server-for-browser-selection), whereas others are [Browserstack-specific capabilities](https://www.browserstack.com/automate/capabilities).

File: .circleci/config.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (kevinbarabash) [reply .circleci/config.yml:114]: I thought that `when` and `unless` we simply the names for the steps, it sounds like that's not the case.  I couldn't find any mention of them in the docs for GitHub actions.

**Comment by ronkok** [comment]
Hey, this is really cool, but I have a question. I use CircleCI to create screenshotter images and CircleCI does not create Safari images that I know of. Is there a way, short of Docker, for me to create Safari screenshotter images?

**Comment by ylemkimon** [comment]
@ronkok You can run (assuming Windows):
```bat
set BROWSERSTACK_USER=(Browserstack username)
set BROWSERSTACK_ACCESS_KEY=(Browserstack access key)
yarn node dockers/screenshotter/screenshotter.js -b safari --browserstack --selenium-capabilities "{""browserName"": ""Safari"", ""browser_version"": ""13.1"", ""os"": ""OS X"", ""os_version"": ""Catalina""}"
```

**Comment by ronkok** [comment]
Is `BROWSERSTACK_ACCESS_KEY` something that one acquires for ($29/month)[https://www.browserstack.com/pricing]?

**Comment by ylemkimon** [comment]
@ronkok No, we're on the open source plan. I've sent invitations after #1879, but if you haven't received it, please let me know the email address. :smile:

**Comment by ronkok** [comment]
@ylemkimon Thank you. My email is ronkok55 {at} outlook.com

**Comment by ylemkimon** [comment]
@ronkok Done! I've deleted your comment for privacy.

**Pull request merged**